File: src/main/java/dev/slne/discord/exception/command/pre/PreCommandCheckException.java
package dev.slne.discord.exception.command.pre;

import dev.slne.discord.exception.DiscordException;
import java.io.Serial;

public class PreCommandCheckException extends DiscordException {

  @Serial
  private static final long serialVersionUID = 129434452849941329L;

  public PreCommandCheckException(String message) {
    super(message);
  }

  public PreCommandCheckException(String message, Throwable cause) {
    super(message, cause);
  }

  public PreCommandCheckException(DiscordException exception) {
    super(exception);
  }
}

File: src/main/kotlin/dev/slne/discord/exception/command/pre/PreCommandCheckException.kt
package dev.slne.discord.exception.command.pre

import dev.slne.discord.exception.DiscordException
import java.io.Serial

open class PreCommandCheckException : DiscordException {
    constructor(message: String?) : super(message)

    constructor(message: String?, cause: Throwable?) : super(message, cause)

    constructor(exception: DiscordException) : super(exception)

    companion object {
        @Serial
        private const val serialVersionUID = 129434452849941329L
    }
}
